### PR TITLE
Added Rust missing-number solution

### PR DESCRIPTION
**Pull request by jxs**

### CODE REVIEW

File: problems/missing-number/missing-number.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  let mut numbers = Vec::from(numbers);
+  numbers.sort();
+  let sum: isize = numbers.iter().sum();
+  let incremental_sum = ((numbers.len() + 1) as isize * (numbers.last()? + numbers.get(0)?)) / 2;
+  Some(incremental_sum - sum)
+}
+
+
+fn main() {
+  println!("missing number: {}", missing_number(&[1,3,5,7,11]).unwrap());
```
Comment (Fleshgrinder) [inline problems/missing-number/missing-number.rs]: This gives 9 but 2, 4, 6, 8, 9, and 10 are missing. The input is actually incorrect. I believe that all solutions suffer from a misinterpretation of the problem.

**Review by Fleshgrinder — CHANGES_REQUESTED** [review]

**Review by jxs — COMMENTED** [review]
sorry didn't see this notification until now. Updated!

**Comment by mre** [comment]
The additional tests were nice. Why did you remove them? 🤔

**Comment by jxs** [comment]
> The additional tests were nice. Why did you remove them? 🤔

cause the after reading https://github.com/mre/the-coding-interview/pull/113#discussion_r352460062 and following the new implementation from other solutions most of the tests were failing. Added back the passing ones and made the implementation only accept `usize`'s as some tests were for negative ranges.

**Comment by mre** [comment]
Thanks for your contribution @jxs and sorry that it took very long to get merged.

**Pull request merged**